Protocol/Tests/Protocol/QActions/QAction/CheckFileEncoding.cs

```diff
@@ -0,0 +1,116 @@
+namespace Skyline.DataMiner.CICD.Validators.Protocol.Tests.Protocol.QActions.QAction.CheckFileEncoding
+{
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+
+    using Skyline.DataMiner.CICD.FileSystem;
+    using Skyline.DataMiner.CICD.Models.Protocol;
+    using Skyline.DataMiner.CICD.Models.Protocol.Read;
+    using Skyline.DataMiner.CICD.Parsers.Common.VisualStudio.Projects;
+    using Skyline.DataMiner.CICD.Validators.Common.Interfaces;
+    using Skyline.DataMiner.CICD.Validators.Common.Model;
+    using Skyline.DataMiner.CICD.Validators.Protocol.Common;
+    using Skyline.DataMiner.CICD.Validators.Protocol.Common.Attributes;
+    using Skyline.DataMiner.CICD.Validators.Protocol.Common.Extensions;
+    using Skyline.DataMiner.CICD.Validators.Protocol.Interfaces;
+
+    [Test(CheckId.CheckFileEncoding, Category.QAction)]
+    internal class CheckFileEncoding : IValidate, ICodeFix/*, ICompare*/
+    {
+        public List<IValidationResult> Validate(ValidatorContext context)
+        {
+            List<IValidationResult> results = new List<IValidationResult>();
+
+            if (!context.HasQActionsAndIsSolution)
+            {
+                // Early skip when no QActions are present or when it is not solution based.
+                return results;
+            }
+
+            foreach ((CompiledQActionProject projectData, IQActionsQAction qaction) in context.EachQActionProject(true))
+            {
+                Project project = Project.Load(projectData.Project.FilePath, projectData.Project.Name);
```

Comment:
can throw a FileNotFoundException in case a cs file mentioned in the csproj is not found.